charts/cortex/templates/store-gateway/store-gateway-statefulset.yaml

```diff
@@ -129,4 +130,152 @@ spec:
           env:
             {{- toYaml .Values.store_gateway.env | nindent 12 }}
           {{- end }}
-{{- end -}}
+{{- end }}
+{{- if .Values.topologyAware.enabled }}
+{{- range $availabilityCell := .Values.topologyAware.availabilityCells }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "cortex-store-gateway.fullname" $ }}-{{ $availabilityCell }}
+  labels: {{ include "cortex-store-gateway.labels" $ | nindent 4 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 4 }}: {{ $availabilityCell }}
+  annotations:
+    {{- toYaml $.Values.store_gateway.annotations | nindent 4 }}
+spec:
+  replicas: {{ $.Values.store_gateway.replicas }}
+  selector:
+    matchLabels: {{ include "cortex-store-gateway.matchLabels" $ | nindent 6 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 6 }}: {{ $availabilityCell }}
+  updateStrategy:
+    {{- toYaml $.Values.store_gateway.strategy | nindent 4 }}
+  serviceName: {{ template "cortex-store-gateway.headlessname" $ }}
```

Comment:
I think we need as many headless services as there are statefulsets - just as with the ingester.